### PR TITLE
AYS-242 | Role Activation Flow Has Been Created

### PR DESCRIPTION
**Pull request by egehanasal**
### Checklist

Before submitting your pull request, ensure the following:

- [X] **Title and Branch Naming Conventions:**
    - The pull request title follows the
      standard: [Pull Request Naming Conventions](CONTRIBUTING.md#pull-request-naming-conventions).
    - The branch name follows one of the conventions outlined in
      the [Branch Naming Conventions](CONTRIBUTING.md#branch-naming-conventions).

- [X] **Local Testing:**
    - I have tested my changes locally on Postman, and they are working as expected.

- [X] **Code Quality:**
    - The code is formatted according to the project's coding guidelines and style.
    - The code has been reviewed to ensure its quality.
    - The code does not contain any issues flagged by SonarLint.

- [X] **Documentation:**
    - Necessary documentation has been added or existing documentation has been updated, specifically detailing changes made in Postman.

- [X] **Testing:**
    - Relevant unit tests have been written and included.
    - Relevant integration tests have been written and included.

- [X] **Reviewers and Assignees:**
    - Default reviewers have been assigned to this pull request.
    - Assignees have been added if necessary.

- [X] **Labels and Associations:**
    - No specific actions are required in the Labels and Associations section for this pull request.

### CODE REVIEW

File: src/main/java/org/ays/auth/service/impl/AysRoleUpdateServiceImpl.java [hunk 1]
```diff
@@ -63,6 +64,26 @@ public void update(final String id,
...
+     *
+     * @param id The ID of the role to activate.
+     * @throws AysRoleNotExistByIdException if a role with the given ID does not exist.
+     */
+    @Override
+    public void activate(String id) {
+        final AysRole role = roleReadPort.findById(id)
+                .orElseThrow(() -> new AysRoleNotExistByIdException(id));
+
+        role.setStatus(AysRoleStatus.ACTIVE);
```
Comment (agitrubard) [inline src/main/java/org/ays/auth/service/impl/AysRoleUpdateServiceImpl.java:82]:
```suggestion
        role.activate();
```

File: src/main/java/org/ays/auth/service/impl/AysRoleUpdateServiceImpl.java [hunk 2]
```diff
@@ -63,6 +64,26 @@ public void update(final String id,
         roleSavePort.save(role);
     }
 
+    /**
```
Comment (agitrubard) [inline src/main/java/org/ays/auth/service/impl/AysRoleUpdateServiceImpl.java:67]:
```suggestion


    /**
```

File: src/test/java/org/ays/auth/controller/AysRoleControllerTest.java [hunk 3]
```diff
@@ -660,4 +660,59 @@ void givenValidIdAndInvalidRoleUpdateRequest_whenPermissionIdIsNotValid_thenRetu
                 .update(Mockito.anyString(), Mockito.any(AysRoleUpdateRequest.class));
     }
 
+
+    @Test
+    void givenValidRoleId_whenRoleActivated_thenReturnSuccess() throws Exception {
```
Comment (agitrubard) [inline src/test/java/org/ays/auth/controller/AysRoleControllerTest.java:665]:
```suggestion
    void givenValidId_whenRoleActivated_thenReturnSuccess() throws Exception {
```

File: src/main/java/org/ays/auth/service/impl/AysRoleUpdateServiceImpl.java [hunk 4]
```diff
@@ -63,6 +64,26 @@ public void update(final String id,
...
+     * an exception is thrown.
+     * </p>
+     *
+     * @param id The ID of the role to activate.
+     * @throws AysRoleNotExistByIdException if a role with the given ID does not exist.
+     */
+    @Override
+    public void activate(String id) {
+        final AysRole role = roleReadPort.findById(id)
+                .orElseThrow(() -> new AysRoleNotExistByIdException(id));
```
Comment (agitrubard) [inline src/main/java/org/ays/auth/service/impl/AysRoleUpdateServiceImpl.java:80]: Rol'ün pasif olmadığı için güncellenemediğine dair bir hata fırlatmamız iyi olabilir, çünkü pasif olmayan rolleri aktifleştirememeliyiz

File: src/test/java/org/ays/util/AysMockMvcRequestBuilders.java [hunk 5]
```diff
@@ -66,6 +66,12 @@ public MockHttpServletRequestBuilder delete(String endpoint, String token, Objec
                 .contentType(MediaType.APPLICATION_JSON);
     }
 
+    public MockHttpServletRequestBuilder patch(String endpoint, String token) {
+        return MockMvcRequestBuilders.patch(endpoint)
+                .header(HttpHeaders.AUTHORIZATION, getTokenWithBearerPrefix(token))
+                .contentType(MediaType.APPLICATION_JSON);
+    }
```
Comment (agitrubard) [inline src/test/java/org/ays/util/AysMockMvcRequestBuilders.java:73]: Bu metodu `delete` metotlarının üstüne taşıyabiliriz

File: src/test/java/org/ays/auth/controller/AysRoleEndToEndTest.java [hunk 6]
```diff
@@ -411,4 +413,56 @@ void givenValidRoleUpdateRequest_whenRoleUpdated_thenReturnSuccess() throws Exce
...
+                        .doesNotExist());
+
+        // Verify
+        Optional<AysRole> roleFromDatabase = roleReadPort.findById(id);
+
+        Assertions.assertTrue(roleFromDatabase.isPresent());
+        Assertions.assertEquals(roleFromDatabase.get().getId(), id);
+        Assertions.assertEquals(AysRoleStatus.ACTIVE, roleFromDatabase.get().getStatus());
+    }
+
```
Comment (agitrubard) [inline src/test/java/org/ays/auth/controller/AysRoleEndToEndTest.java:467]: Burayı servis güncellendikten sonra review edeceğim fakat burada id'nin bulunmadığına dair test yok

**Review by agitrubard — CHANGES_REQUESTED** [review]

**Review by MenekseYuncu — APPROVED** [review]

File: src/main/java/org/ays/auth/service/impl/AysRoleUpdateServiceImpl.java (hunk 1 above)
Comment (egehanasal) [reply src/main/java/org/ays/auth/service/impl/AysRoleUpdateServiceImpl.java]: değişikliği yaptım

File: src/main/java/org/ays/auth/service/impl/AysRoleUpdateServiceImpl.java (hunk 2 above)
Comment (egehanasal) [reply src/main/java/org/ays/auth/service/impl/AysRoleUpdateServiceImpl.java]: değişikliği yaptım

File: src/test/java/org/ays/auth/controller/AysRoleControllerTest.java (hunk 3 above)
Comment (egehanasal) [reply src/test/java/org/ays/auth/controller/AysRoleControllerTest.java]: değişikliği yaptım

File: src/main/java/org/ays/auth/service/impl/AysRoleUpdateServiceImpl.java (hunk 4 above)
Comment (egehanasal) [reply src/main/java/org/ays/auth/service/impl/AysRoleUpdateServiceImpl.java:86]: Güncelledim.

File: src/test/java/org/ays/util/AysMockMvcRequestBuilders.java (hunk 5 above)
Comment (egehanasal) [reply src/test/java/org/ays/util/AysMockMvcRequestBuilders.java]: değişikliği yaptım

File: src/test/java/org/ays/auth/controller/AysRoleEndToEndTest.java (hunk 6 above)
Comment (egehanasal) [reply src/test/java/org/ays/auth/controller/AysRoleEndToEndTest.java:458]: Testleri güncelledim

**Review by agitrubard — CHANGES_REQUESTED** [review]

File: src/main/java/org/ays/auth/service/impl/AysRoleUpdateServiceImpl.java [hunk 7]
```diff
@@ -65,6 +67,33 @@ public void update(final String id,
...
+     * @param id The ID of the role to activate.
+     * @throws AysRoleNotExistByIdException if a role with the given ID does not exist.
+     * @throws AysInvalidRoleStatusException if the role's current status is not {@link AysRoleStatus#PASSIVE}.
+     */
+    @Override
+    public void activate(String id) {
+        final AysRole role = roleReadPort.findById(id)
+                .orElseThrow(() -> new AysRoleNotExistByIdException(id));
+
+        if(role.getStatus() != AysRoleStatus.PASSIVE) {
```
[the remark this answers is not in the document]
Comment (egehanasal) [reply src/main/java/org/ays/auth/service/impl/AysRoleUpdateServiceImpl.java]: Değiştirdim

File: src/main/java/org/ays/auth/model/AysRole.java [hunk 8]
```diff
@@ -48,6 +48,14 @@ public boolean isPassive() {
         return AysRoleStatus.PASSIVE.equals(this.status);
     }
 
+    /**
+     * Activates the role by setting its status to {@link AysRoleStatus#ACTIVE}.
+     * This method should be called when the role needs to be marked as active in the system.
+     */
+    public void activate(){
```
[the remark this answers is not in the document]
Comment (egehanasal) [reply src/main/java/org/ays/auth/model/AysRole.java]: teşekkürler. Düzelttim

**Review by agitrubard — APPROVED** [review]

**Review by ybilgaz96 — APPROVED** [review]

**Pull request merged**